Superbuild failure in spotless, when matlab is not installed

From @jwnimmer-tri:  a clean superbuild with `env PATH=/usr/lib/ccache:$PATH CXX=g++-4.9 CC=gcc-4.9 make options (ceceg) && env PATH=/usr/lib/ccache:$PATH CXX=g++-4.9 CC=gcc-4.9 make -j8` randomly dies in superbuild externals for no discernable reason.  I eventually just gave up, did it under -j1, and now just use -j8 in drake.

Here's my steps on Ubuntu 14.04 without matlab installed:
 ```
jwnimmer@call:~/tmp$ git clone https://github.com/RobotLocomotion/drake.git drake-distro
jwnimmer@call:~/tmp$ cd drake-distro/
jwnimmer@call:~/tmp/drake-distro$ env CXX=g++-4.9 CC=gcc-4.9 make options  #ceceg
Configuring with CMAKE_FLAGS: -DCMAKE_BUILD_TYPE=Release -DPOD_BUILD=TRUE
...
-- CMAKE_INSTALL_PREFIX=/home/jwnimmer/tmp/drake-distro/build
-- Found Java: /usr/bin/java (found version "1.8.0.77") 
Looked for MATLAB but could not find it.
-- Preparing to build cmake with dependencies: 
...
-- Preparing to build spotless with dependencies: 
...
-- Preparing to build drake with dependencies: cmake;eigen;googletest;lcm;bot_core_lcmtypes;libbot;bullet;swig_matlab;swigmake;yaml_cpp
...
-- Build files have been written to: /home/jwnimmer/tmp/drake-distro/pod-build
jwnimmer@call:~/tmp/drake-distro$ env CXX=g++-4.9 CC=gcc-4.9 make download-all
Configuring with CMAKE_FLAGS: -DCMAKE_BUILD_TYPE=Release -DPOD_BUILD=TRUE
-- CMAKE_INSTALL_PREFIX=/home/jwnimmer/tmp/drake-distro/build
Looked for MATLAB but could not find it.
...
-- Preparing to build spotless with dependencies: 
...
jwnimmer@call:~/tmp/drake-distro$ env CXX=g++-4.9 CC=gcc-4.9 make -j8
...
[ 70%] CMake Error at CMakeLists.txt:16 (message):
  Could not find matlab executable
...
[ 11%] make[3]: *** [spotless-prefix/src/spotless-stamp/spotless-configure] Error 1
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,6 @@ endif()
 
 include(CMakeDependentOption)
 option(DISABLE_MATLAB "Don't use MATLAB even if it is present." OFF)
-CMAKE_DEPENDENT_OPTION(WITH_SPOTLESS "polynomial optimization front-end for MATLAB" ON "NOT DISABLE_MATLAB" OFF)
 if (DISABLE_MATLAB)
 	message(STATUS "MATLAB is disabled.")
 else()
@@ -84,9 +83,13 @@ else()
 	if (matlab)
 		message(STATUS "Found MATLAB at " ${matlab})
 	else ()
-		message("Looked for MATLAB but could not find it.")
+		message(STATUS "Looked for MATLAB but could not find it.")
 	endif(matlab)
 endif (DISABLE_MATLAB)
+
+# If DISABLE_MATLAB is set or matlab is not found, WITH_SPOTLESS is hidden and default OFF.
+# Otherwise, WITH_SPOTLESS is visible and default ON.
+CMAKE_DEPENDENT_OPTION(WITH_SPOTLESS "polynomial optimization front-end for MATLAB" ON "NOT DISABLE_MATLAB;matlab" OFF)
 
 option(WITH_YAML_CPP "library for reading and writing yaml configuration files" ON)
 

